x-pack/plugins/ml/public/explorer/explorer.js

```diff
@@ -52,48 +114,782 @@ function mapSwimlaneOptionsToEuiOptions(options) {
+
+    dragSelect = new DragSelect({
+      selectables: document.getElementsByClassName('sl-cell'),
+      callback(elements) {
+        if (elements.length > 1 && !this.ALLOW_CELL_RANGE_SELECTION) {
+          elements = [elements[0]];
+        }
+
+        if (elements.length > 0) {
+          mlExplorerDashboardService.dragSelect.changed({
+            action: DRAG_SELECT_ACTION.NEW_SELECTION,
+            elements
+          });
+        }
+
+        this.disableDragSelectOnMouseLeave = true;
+      },
+      onDragStart() {
+        if (this.ALLOW_CELL_RANGE_SELECTION) {
+          mlExplorerDashboardService.dragSelect.changed({
+            action: DRAG_SELECT_ACTION.DRAG_START
+          });
+          this.disableDragSelectOnMouseLeave = false;
+        }
+      },
+      onElementSelect() {
+        if (this.ALLOW_CELL_RANGE_SELECTION) {
+          mlExplorerDashboardService.dragSelect.changed({
+            action: DRAG_SELECT_ACTION.ELEMENT_SELECT
+          });
+        }
+      }
+    });
+
+    dashboardListener = (async (action, payload) => {
+      // Listen to the initial loading of jobs
+      if (action === EXPLORER_ACTION.INITIALIZE) {
+        const {
+          appState,
+          jobs,
+          mlJobSelectService
+        } = this.props;
+        // Select any jobs set in the global state (i.e. passed in the URL).
+        const selectedJobIds = mlJobSelectService.getSelectedJobIds(true);
+
+        const selectedJobs = await getSelectedJobs(this.state.selectedJobs, jobs, selectedJobIds, appState);
+        this.setState(
+          {
+            noInfluencersConfigured: !selectedJobsHaveInfluencers(selectedJobs),
+            selectedJobs
+          },
+          async () => {
+            // Load the data - if the FieldFormats failed to populate
+            // the default formatting will be used for metric values.
+            await this.loadOverallData();
+            this.loadViewBySwimlane([]);
+
+            let selectedCells = this.state.selectedCells;
+
+            // keep swimlane selection, restore selectedCells from AppState
+            if (
+              selectedCells === undefined &&
+              appState.mlExplorerSwimlane.selectedType !== undefined
+            ) {
+              selectedCells = {
+                type: appState.mlExplorerSwimlane.selectedType,
+                lanes: appState.mlExplorerSwimlane.selectedLanes,
+                times: appState.mlExplorerSwimlane.selectedTimes
+              };
+              if (selectedCells.type === SWIMLANE_TYPE.VIEW_BY) {
+                selectedCells.fieldName = appState.mlExplorerSwimlane.viewBy;
+              }
+              this.setState(
+                {
+                  selectedCells,
+                  swimlaneViewByFieldName: appState.mlExplorerSwimlane.viewBy,
+                },
+                () => {
+                  this.updateExplorer();
+                }
```

Comment:
I don't think this needs to be wrapped in an arrow function since you're not passing any arguments to it.

Reply:
Fixed in 51fe618b4c (not present as is anymore because of an outer refactore).